Makefile

```diff
@@ -100,14 +104,36 @@ clean:
 clean-cache:
 	rm -rf ${CACHEDIR}
 
-${CACHEDIR}/.libraries_token.txt: cache
-	tmpf=$(shell mktemp); \
-	chainctl auth login --audience libraries.cgr.dev; \
-	chainctl auth token --audience libraries.cgr.dev > $${tmpf}; \
-	mv $${tmpf} ${CACHEDIR}/.libraries_token.txt
-
-.PHONY: lib-token
-lib-token: ${CACHEDIR}/.libraries_token.txt
+%/.guarded-repo.token:
+	$(eval pkgname := $*)
+	tmpf=$(shell mktemp) && \
+	chainctl auth octo-sts --identity=guarded-package-repos --scope=chainguard-dev > $${tmpf} && \
+	mkdir -p $(pkgname) && \
+	mv $${tmpf} ./$(pkgname)/.guarded-repo.token
+
+repo_token_targets = $(foreach name,$(pkgs),repo-token/$(name))
+$(repo_token_targets): repo-token/%: %/.guarded-repo.token
+
+%/.libraries.token:
+	$(eval pkgname := $*)
+	tmpf=$(shell mktemp) && \
+	chainctl auth token --audience libraries.cgr.dev > $${tmpf} && \
+	mkdir -p $(pkgname) && \
+	mv $${tmpf} ./$(pkgname)/.libraries.token
+
+libraries_token_targets = $(foreach name,$(pkgs),lib-token/$(name))
+$(libraries_token_targets): lib-token/%: %/.libraries.token
+
+.PHONY: tokens-if-needed/%
```

Comment:
I don't see where this rule is defined, but maybe I'm just tired :/

Reply:
Fixed this, my bad